lib/simctl.js

```diff
@@ -44,15 +45,10 @@ async function simCommand (command, timeout, args = [], env = {}, executingFunct
   try {
     return await executingFunction('xcrun', args, {timeout, env});
   } catch (e) {
-    if (!logErrors) {
-      // if we don't want to see the errors, just throw and allow the calling
-      // code do what it wants
-      throw e;
-    } else if (e.stderr) {
-      log.errorAndThrow(`simctl error running '${command}': ${e.stderr.trim()}`);
-    } else {
-      log.errorAndThrow(e);
+    if (logErrors && e.stderr) {
```

Comment:
This seems like a change in the semantics of `logErrors`. The docstring says `"Set it to _false_ to throw execution errors immediately without logging any additional information"`, which does not any longer seem to be the case.

Reply:
makes sense

In general, I observe a lot of redundancy in this module. I'd rather make it into a class, so all these functions don't have to accept the same set of arguments (like udid and/or logErrors), but rather accept them in the constructor. What do you think @imurchie ?

Reply:
I agree. Especially since it seems like someone is actually adding things to `simctl`.